### PR TITLE
fix: Nonce has no `from` other than trivial one

### PR DESCRIPTION
**Pull request by giladchase**

### CODE REVIEW

**Review by ayeletstarkware — COMMENTED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/mempool/342)* status: 0 of 1 files reviewed, 1 unresolved discussion (waiting on @elintul and @giladchase)

___
*[`crates/mempool/src/mempool_test.rs` line 28 at r1](https://reviewable.io/reviews/starkware-libs/mempool/342#-O0ms5lp91AADnruDh_d:-O0ms5lp91AADnruDh_e:bie0ouu) ([raw file](https://github.com/starkware-libs/mempool/blob/e51ec097519d465471bf6db5ef6e2a1c60472052/crates/mempool/src/mempool_test.rs#L28)):*
> ```XML
>             tx_hash: TransactionHash($tx_hash),
>             sender_address,
>             nonce: Nonce::from($nonce),
> ```

Can you explain the difference? Is it only for improving readability?



_Code quote:_
```XML
Nonce::from($nonce),
```

**Review by giladchase — COMMENTED** [review]
> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/mempool/342)* status: all files reviewed, 1 unresolved discussion (waiting on @ayeletstarkware)

___
*[`crates/mempool/src/mempool_test.rs` line 28 at r1](https://reviewable.io/reviews/starkware-libs/mempool/342#-O0ms5lp91AADnruDh_d:-O0muL-O2DYOyG2nekjl:bb4dqn1) ([raw file](https://github.com/starkware-libs/mempool/blob/e51ec097519d465471bf6db5ef6e2a1c60472052/crates/mempool/src/mempool_test.rs#L28)):*
<details><summary><i>Previously, ayeletstarkware (Ayelet Zilber) wrote…</i></summary><blockquote>

Can you explain the difference? Is it only for improving readability?
</blockquote></details>

Nonce currently has no From implementations, other than the trivial "blanket implementation" of from (google the term :grimacing: or "go to definition" on the existing `from`).

So basically that `from` was a lie, and I need to use the constructor, which is all I have to use really (alternatively I can Implement a From in SNAPI but that'd be weird, since no one would want to create a nonce this way outside of tests).

**Review by ayeletstarkware — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/mempool/342)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @giladchase)

**Review by giladchase — COMMENTED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/mempool/342)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @giladchase)
